src/js/components/chat-input.jsx

```diff
@@ -38,9 +38,17 @@ export class ChatInputComponent extends Component {
 
     resizeInput(attempt = 0) {
         const node = findDOMNode(this);
-        if (node.offsetWidth - this.refs.button.offsetWidth > 0) {
+
+        const nodeRect = node.getBoundingClientRect();
+        const buttonRect = this.refs.button.getBoundingClientRect();
+
+        // use floor on widget and ceil on button to get worst rounding scenario
```

Comment:
find the minimum width the widget could take when it's a decimal, find the maximum width the button can take when it's a decimal.. that way, widget width - button width is always smaller or equal than the actual width